components/sysutils/p0f/patches/001-p0f-3.09b-solarisfix.patch

```diff
@@ -0,0 +1,859 @@
+Fixes against p0f-3.06b (updated for 3.09b) to have it compile and work
+properly under Solaris/illumos. Also integrates with Solaris 10+ SMF serices.
+Fixes licensed under the terms of MIT License.
```

Comment:
Hmmm... that is an interesting point, perhaps the patch-files should be split with varied licensing. E.g. I have no preference for my own code (SMF manifest and wrapper script), so it could be MIT or BSD includable by everybody. There are some fixes to original build script and C code that would indeed follow LGPL2 (although again my fixes posted to the world under MIT license should be includable and redistributable by any-license code including (L)GPL) : http://stackoverflow.com/questions/3902754/mit-vs-gpl-license

The hardest bit is about a few trivial but useful routines snatched from the internet or Solaris sources, concentrated in `p0f.c`: the `flock()` wrapper which originates from NetBSD toolchain (see http://www.perkin.org.uk/posts/solaris-portability-flock.html for details) which should be includable into GPL too - e.g. http://opensource.stackexchange.com/questions/304/can-i-take-bsd-licensed-code-and-distribute-it-under-gpl discusses that.

Then there is also a `strioctl()` wrapper that I took IIRC from Solaris snoop (so maybe it is CDDL) which is IMHO too trivial to be a contention point - the same setup of structure and call to `ioctl()` is often done "inline" (e.g. shown in https://illumos.org/man/7m/timod manpage), it just looks cumbersome and repetitive to do so. 

Example usage of those ioctl's later in the file, that got me started, originates from wpa-supplicant and nmap (both BSD licensed).

So I guess the wording can be changed that I post my patch under BSD license, so there is less confusion (I can't well tell MIT and BSD apart in practice), and it should be bindable with a GPL project. One oddball is the ioctl wrapper, maybe...